core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -30,7 +30,7 @@ class InterBrokerSendThread(name: String,
                             networkClient: NetworkClient,
                             requestGenerator: () => Iterable[RequestAndCompletionHandler],
                             time: Time)
-  extends ShutdownableThread(name, isInterruptible = false) {
+  extends ShutdownableThread(name, isInterruptible = true) {
```

Comment:
I felt it is safer to not make the thread interruptible but wait for it to complete the current iteration and exit. The reason being this thread may be used to replace replica fetcher thread and controller channel sender thread. And for the replica fetcher thread it involves IO operations.

Or we can at least make this overriable and let the `txnMarkerSender` to override it to true.